From fe6e62382c45089da1b2677e63acb88064fd2ac0 Mon Sep 17 00:00:00 2001
From: Daniel Leung <daniel.leung@intel.com>
Date: Wed, 6 Feb 2013 16:11:01 -0800
Subject: Use GCC 4.7 for x86 targets

Issue: AXIA-1852
Change-Id: If2abe386f2203a8f37c08eb70738277c2eab0b04
Depends-Change-Id: I3e8556046b00f2bd197e63d39500a345abc5a6e4
Depends-Change-Id: I8cd239880821724050d1716b78851807e0246ef2
Depends-Change-Id: I1d1434bc884796d5c862aff17f7b823f0d7ca59d
Signed-off-by: Daniel Leung <daniel.leung@intel.com>
---
 core/combo/arch/x86/haswell.mk   |    4 +---
 core/combo/arch/x86/ivybridge.mk |    2 +-
 envsetup.sh                      |    4 ++--
 3 files changed, 4 insertions(+), 6 deletions(-)

diff --git a/core/combo/arch/x86/haswell.mk b/core/combo/arch/x86/haswell.mk
index 45c8e29..3f98047 100644
--- a/core/combo/arch/x86/haswell.mk
+++ b/core/combo/arch/x86/haswell.mk
@@ -1,7 +1,5 @@
 # Configuration for Linux on x86.
 # Generating binaries for Haswell processors.
-# FIXME: This is just ivy bridge; update for Haswell
-# capabilities once we have compiler support
 
 ARCH_X86_HAVE_MMX    := true
 ARCH_X86_HAVE_SSE    := true
@@ -16,7 +14,7 @@ ARCH_X86_HAVE_AVX    := true
 
 # CFLAGS for this arch
 arch_variant_cflags := \
-	-march=corei7-avx \
+	-march=core-avx2 \
 	-mstackrealign \
 	-mfpmath=sse \
 
diff --git a/core/combo/arch/x86/ivybridge.mk b/core/combo/arch/x86/ivybridge.mk
index 32b957f..c0f8d89 100644
--- a/core/combo/arch/x86/ivybridge.mk
+++ b/core/combo/arch/x86/ivybridge.mk
@@ -14,7 +14,7 @@ ARCH_X86_HAVE_AVX    := true
 
 # CFLAGS for this arch
 arch_variant_cflags := \
-	-march=corei7-avx \
+	-march=core-avx-i \
 	-mstackrealign \
 	-mfpmath=sse \
 
diff --git a/envsetup.sh b/envsetup.sh
index 93a4ab7..c63aa58 100644
--- a/envsetup.sh
+++ b/envsetup.sh
@@ -150,7 +150,7 @@ function setpaths()
                  ARM_EABI_TOOLCHAIN_PATH=":$gccprebuiltdir/$toolchaindir"
             fi
             ;;
-        x86) toolchaindir=x86/i686-linux-android-4.6/bin
+        x86) toolchaindir=x86/i686-linux-android-4.7/bin
             ;;
         mips) toolchaindir=mips/mips-eabi-4.4.3/bin
             ;;
@@ -160,7 +160,7 @@ function setpaths()
     esac
 
     export ANDROID_TOOLCHAIN=$ANDROID_EABI_TOOLCHAIN
-    export X86_TOOLCHAIN=$prebuiltdir/toolchain/i686-linux-android-4.6/bin
+    export X86_TOOLCHAIN=$prebuiltdir/toolchain/i686-linux-android-4.7/bin
     export ANDROID_QTOOLS=$T/development/emulator/qtools
     export ANDROID_DEV_SCRIPTS=$T/development/scripts:$T/prebuilts/devtools/tools
     export ANDROID_BUILD_PATHS=$(get_build_var ANDROID_BUILD_PATHS):$ANDROID_QTOOLS:$ANDROID_TOOLCHAIN$ARM_EABI_TOOLCHAIN_PATH$CODE_REVIEWS:$X86_TOOLCHAIN:$ANDROID_DEV_SCRIPTS:
-- 
1.7.4.1

